core/src/main/java/com/linecorp/armeria/server/CorsServerErrorHandler.java

```diff
@@ -50,49 +47,19 @@ public AggregatedHttpResponse renderStatus(@Nullable ServiceRequestContext ctx,
                                                @Nullable RequestHeaders headers,
                                                HttpStatus status, @Nullable String description,
                                                @Nullable Throwable cause) {
-
-        if (ctx == null) {
-            return serverErrorHandler.renderStatus(null, serviceConfig, headers, status, description, cause);
-        }
-
-        final CorsService corsService = ctx.findService(CorsService.class);
-        if (corsService == null) {
-            return serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status, description, cause);
-        }
-
-        final AggregatedHttpResponse res = serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status,
-                                                                           description, cause);
-
-        if (res == null) {
-            return serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status, description, cause);
-        }
-
-        final CorsConfig corsConfig = corsService.config();
-        final ResponseHeaders updatedResponseHeaders = addCorsHeaders(ctx, corsConfig,
-                                                                      res.headers());
-
-        return AggregatedHttpResponse.of(updatedResponseHeaders, res.content());
+        return serverErrorHandler.renderStatus(ctx, serviceConfig, headers, status, description, cause);
     }
 
     @Nullable
     @Override
     public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
-        if (cause instanceof HttpResponseException) {
-            final HttpResponse oldRes = serverErrorHandler.onServiceException(ctx, cause);
-            if (oldRes == null) {
-                return null;
-            }
-            final CorsService corsService = ctx.findService(CorsService.class);
-            if (corsService == null) {
-                return oldRes;
-            }
-            return oldRes.recover(HttpResponseException.class, ex -> {
-                return ex.httpResponse()
-                         .mapHeaders(oldHeaders -> addCorsHeaders(ctx, corsService.config(), oldHeaders));
+        final CorsService corsService = ctx.findService(CorsService.class);
```

Comment:
Note: This is also a similar issue, but a debug level log will be printed every time an exception is thrown for an Origin which doesn't have a policy defined:

https://github.com/line/armeria/blob/683bd74b91eec18a1a5007c4ac1013327e7b5a62/core/src/main/java/com/linecorp/armeria/internal/server/CorsHeaderUtil.java#L127

Normally, this log wouldn't be printed if a request goes through `CorsService`